drivers/st77xx/Kconfig.st7735

```
@@ -0,0 +1,16 @@
+# Copyright (c) 2023 Gunar Schorcht
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+if MODULE_ST77XX
+
+config MODULE_ST7735
+    bool "ST7735 display"
+    default y if !MODULE_ST7789 && !MODULE_ST7796
```

Why not use the following?
```suggestion
    default y if HAVE_ST7735
```
like for other variants.

---

Just to model the dependencies exactly as defined in the makefiles: https://github.com/RIOT-OS/RIOT/blob/2c0c1caa33158436b6906c441f34d2f279b33e94/drivers/st77xx/Makefile.dep#L9-L13 Do you think that your suggestion would also give the same results. If so, I'm fine with it.

---

I don't think it would result in the same.

If I am understanding this modelling correctly the following statements would be made:
- Multiple variants of ST77xx can be explicitly selected
- Selecting ST77xx would default to ST7735 unless a board declares a variant with HAVE_

Using the suggestion would mean that boards would not have a default and require explicit  selection or `HAVE_ST7735`.

If only one variant can be used (due to implementation constraints) then probably better to use a choice, this makes the default more scalable.

---

> If only one variant can be used (due to implementation constraints) then probably better to use a choice, this makes the default more scalable.

It is intentionally a selection. I don't know if it really makes sense or there is a use case where a board has two or more displays with different ST77xx variants. But to make this possible in principle, there can be multiple ST77xx displays and the user can select multiple variants.

---

> If I am understanding this modelling correctly the following statements would be made:
> 
> * Multiple variants of ST77xx can be explicitly selected
> * Selecting ST77xx would default to ST7735 unless a board declares a variant with HAVE_

Exactly.

---

> @aabadie Are you fine now with the Kconfig dependency modelling? This was the last point open for this PR.

I'm fine with the changes. Let's get this in!